src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.ScopedCss.targets

```diff
@@ -103,12 +103,12 @@ Integration with static web assets:
   </DiscoverDefaultScopedCssItems>
 
   <ItemGroup Condition="'$(EnableDefaultScopedCssItems)' == 'true'">
-    <ScopedCssInput Include="@(_DiscoveredScopedCssInputs)" />
+    <ScopedCssInput Include="@(_DiscoveredScopedCssInputs)" Pack="false" CopyToPublishDirectory="Never" />
   </ItemGroup>
 
   <ItemGroup>
-    <Content Remove="@(ScopedCssInput)" />
-    <Content Include="@(ScopedCssInput)" Pack="false" CopyToPublishDirectory="Never" />
+    <!-- Configure Pack="false" and CopyToPublishDirectory="Never" for Content items that are ScopedCssInputs. -->
+    <Content Pack="%(ScopedCssInput.Pack)" CopyToPublishDirectory="%(ScopedCssInput.CopyToPublishDirectory)" />
```

Comment:
@javiercn this seemingly works: https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-items?view=vs-2022#updating-metadata-on-items-in-an-itemgroup-of-a-target. What do you think?

Reply:
demo.msbuildproj

```XML
<Project>
  <ItemGroup>
    <Content Include="A.txt; B.css; C.css" />
    <ScopedCssInput Include="B.css; C.css" Pack="false" CopyToPublishDirectory="Never" />
  </ItemGroup>

  <Target Name="Build">
    <ItemGroup>
      <Content Pack="%(ScopedCssInput.Pack)" CopyToPublishDirectory="%(ScopedCssInput.CopyToPublishDirectory)" />
    </ItemGroup>

    <Message Text="%(Content.Identity): Pack='%(Content.Pack)' CopyToPublishDirectory='%(Content.CopyToPublishDirectory)'" />
  </Target>
</Project>
```

dotnet msbuild demo.msbuildproj -verbosity:detailed

```
Microsoft (R) Build Engine version 17.1.0+ae57d105c for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 23.2.2022 0.00.42.
     0>Process = "C:\Program Files\dotnet\dotnet.exe"
       MSBuild executable path = "C:\Program Files\dotnet\sdk\6.0.200\MSBuild.dll"
       Command line arguments = ""C:\Program Files\dotnet\sdk\6.0.200\dotnet.dll" msbuild demo.msbuildproj -verbosity:detailed"
       Current directory = "C:\Users\Kalle\Desktop"
       MSBuild version = "17.1.0+ae57d105c"
     1>Project "C:\Users\Kalle\Desktop\demo.msbuildproj" on node 1 (default targets).
     1>Target "Build" in project "C:\Users\Kalle\Desktop\demo.msbuildproj" (entry point):
     1>Using "Message" task from assembly "Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
       Task "Message"
         A.txt: Pack='false' CopyToPublishDirectory='Never'
       Done executing task "Message".
       Task "Message"
         B.css: Pack='false' CopyToPublishDirectory='Never'
       Done executing task "Message".
       Task "Message"
         C.css: Pack='false' CopyToPublishDirectory='Never'
       Done executing task "Message".
     1>Done building target "Build" in project "demo.msbuildproj".
     1>Done Building Project "C:\Users\Kalle\Desktop\demo.msbuildproj" (default targets).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.15
```

It applies `Pack="%(ScopedCssInput.Pack)" CopyToPublishDirectory="%(ScopedCssInput.CopyToPublishDirectory)"` even to `A.txt` even though that does not match any `ScopedCssInput` item.

Reply:
I thought that could be fixed by adding `Update="@(ScopedCssInput)"` or `Update="%(ScopedCssInput.Identity)"`, but no, MSBuild ignores that attribute. [This attribute is valid only if it's specified for an item in an `ItemGroup` that is not in a `Target`.](https://docs.microsoft.com/en-us/visualstudio/msbuild/item-element-msbuild?view=vs-2019) Odd that there is not even a warning.

Reply:
Possibly `Condition="'%(Content.Identity)' == '%(ScopedCssInput.Identity)'"` could work, but if it does, I think it will cost quadratic time.

Reply:
If everything else fails, you can just define a custom task for this.

Reply:
Thanks for catching this @KalleOlaviNiemitalo. I tried the positive cases, and was mislead by its apparent success. Adding a condition doesn't work and it'd be easier to resolve this in the MAUI targets where these items are consumed.

Reply:
@pranavkm the way I've done this in the past involves removing the items and re-adding them.